src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaTopicScaler.cs

```diff
@@ -16,14 +16,12 @@ public class KafkaTopicScaler<TKey, TValue> : IScaleMonitor<KafkaTriggerMetrics>
         private readonly string topicName;
         private readonly string consumerGroup;
         private readonly ILogger logger;
-        private readonly AdminClientConfig adminClientConfig;
-        private readonly IConsumer<TKey, TValue> consumer;
-        private readonly Lazy<List<TopicPartition>> topicPartitions;
         private readonly long lagThreshold;
+        private readonly KafkaMetricsProvider<TKey, TValue> metricsProvider;
 
         public ScaleMonitorDescriptor Descriptor { get; }
 
-        public KafkaTopicScaler(string topic, string consumerGroup, string functionId, IConsumer<TKey, TValue> consumer, AdminClientConfig adminClientConfig, long lagThreshold, ILogger logger)
+        internal KafkaTopicScaler(string topic, string consumerGroup, string functionId, IConsumer<TKey, TValue> consumer, KafkaMetricsProvider<TKey, TValue> metricsProvider, long lagThreshold, ILogger logger)
```

Comment:
Have we already released the package with "KafkaTopicScaler"? The class should not be public, also I think a better name is `KafkaScaleMonitor`

Reply:
Yes, we have released package with `KafkaTopicScaler`. First release of KafkaTopicScaler was around version 3.0.0. ([ref PR](https://github.com/Azure/azure-functions-kafka-extension/pull/119)). It has been initially named KafkaTopicScaler as a public class and has been that way so far.
To address its access, I have updated the public access to internal.